packages/dom/src/dom.js

```diff
@@ -1,25 +1,58 @@
 /**
  * External dependencies
  */
-import { includes, first } from 'lodash';
-import tinymce from 'tinymce';
+import { includes } from 'lodash';
 
 /**
  * Browser dependencies
  */
-const { getComputedStyle, DOMRect } = window;
-const { TEXT_NODE, ELEMENT_NODE } = window.Node;
+
+const { getComputedStyle } = window;
+const {
+	TEXT_NODE,
+	ELEMENT_NODE,
+	DOCUMENT_POSITION_PRECEDING,
+} = window.Node;
 
 /**
- * Check whether the caret is horizontally at the edge of the container.
+ * Returns true if the given selection object is in the forward direction, or
+ * false otherwise.
+ *
+ * @see https://developer.mozilla.org/en-US/docs/Web/API/Node/compareDocumentPosition
  *
- * @param {Element} container      Focusable element.
- * @param {boolean} isReverse      Set to true to check left, false for right.
- * @param {boolean} collapseRanges Whether or not to collapse the selection range before the check.
+ * @param {Selection} selection Selection object to check.
+ *
+ * @return {boolean} Whether the selection is forward.
+ */
+function isSelectionForward( selection ) {
+	const {
+		anchorNode,
+		focusNode,
+		anchorOffset,
+		focusOffset,
+	} = selection;
+
+	const position = anchorNode.compareDocumentPosition( focusNode );
+
+	return (
+		// Compare whether anchor node precedes focus node.
+		position !== DOCUMENT_POSITION_PRECEDING &&
+
+		// `compareDocumentPosition` returns 0 when passed the same node, in
+		// which case compare offsets.
+		! ( position === 0 && anchorOffset > focusOffset )
+	);
+}
+
+/**
+ * Check whether the selection is horizontally at the edge of the container.
+ *
+ * @param {Element} container Focusable element.
+ * @param {boolean} isReverse Set to true to check left, false for right.
```

Comment:
This seems like an odd variable name–we aren't setting if something `isReverse`, we're checking for it. Would we want `checkForReverse` or maybe to specify the direction? eg. `isHorizontalEdge( container, direction='left' )`? I know it's a more verbose API but it's also easier to grok. Supplying a binary to this function doesn't provide much context in isolation. (`isHorizontalEdge( container, false )`–what is `false` in that bit?)

Reply:
I'd probably tend to agree, but it's not within scope of the pull request, and since this is a [published module](https://www.npmjs.com/package/@wordpress/dom), such a change would qualify as breaking. Since I'm already touching the documentation, I'll at least try to make the boolean-ness more clear.

Reply:
Yeah, that's fair. I just figured I'd point it out either way. If I still feel strongly about it I'll file a code quality issue later 😉

Reply:
Yes, it's good to point out. I think it's easy to fall into the trap of accepting "simple" values as arguments when authoring the function originally, and not considering that it would be difficult to interpret the intent of the statement `isHorizontalEdge( container, false )`.